lib/pusher/http_client.ex

```diff
@@ -11,7 +11,8 @@ defmodule Pusher.HttpClient do
   end
 
   defp process_response_body(body) do
-    unless body == "", do: body |> JSX.decode!, else: nil
+    :ok = validate_response_body(body)
```

Comment:
The actual reason for the error is contained in the response body (e.g. Invalid auth key).
If we do it the way you propose then it wont be part of the error message.
Thats why i am returning the body as part of the error message in line 36.